src/services/review/ReviewRequestService.ts

```diff
@@ -439,36 +440,74 @@ export default class ReviewRequestService {
     // NOTE: commits from github are capped at 300.
     // This implies that there might possibly be some files
     // whose commit isn't being returned.
-    await Promise.all(
-      commits.map(async ({ commit, sha }) => {
-        const { userId } = fromGithubCommitMessage(commit.message)
-        const author = await this.users.findByPk(userId)
-        const lastChangedTime = new Date(commit.author.date).getTime()
-        mappings[sha] = {
-          author: author?.email || commit.author.name,
-          unixTime: lastChangedTime,
-        }
-      })
+
+    // commitAuthorIds are linked to the commits by array index
+    const commitAuthorIds = commits.map(
+      ({ commit }) => fromGithubCommitMessage(commit.message).userId
     )
+
+    // Query DB for all valid author details
+    const validAuthorIds = new Set(
+      commitAuthorIds.filter(_.identity) as string[]
```

Comment:
comment above, seems weird to type cast when we are not sure if this is a type `string[]`

Reply:
`commit.userId` is types as `string | undefined`, so the `.filter(_.identity)` guarantees  string. I wish TS would see it too 😢

Reply:
The filter also drops empty string too btw, if that ever was an issue 🤷